src/component/Text.js

```diff
@@ -54,12 +54,14 @@ class Text extends Component {
     this.updateWordsByLines(this.props, true);
   }
 
-  componentWillReceiveProps(nextProps) {
-    const needCalculate = (
-      this.props.children !== nextProps.children ||
-      this.props.style !== nextProps.style
-    );
-    this.updateWordsByLines(nextProps, needCalculate);
+  componentDidUpdate(prevProps) {
+    if ((prevProps.width !== this.props.width || prevProps.scaleToFit !== this.props.scaleToFit)) {
```

Comment:
This is the only method i had to change a bit for making it to work. Just some validations for avoiding infinite loop re-render